cpp/support/thread_safe_cache.h

```diff
@@ -179,6 +188,228 @@ class ThreadSafeCache<Key, Value> {
   std::shared_mutex erase_mutex_;
 };
 
+namespace details {
+
+template <typename Value, typename... Keys>
+class LRUCacheImpl {
+ protected:
+  struct LRUnode;
+  using KeySet = std::variant<const Keys*...>;
+  using LRUlist = std::list<LRUnode>;
+  using LRUiterator = typename LRUlist::iterator;
```

Comment:
```suggestion
```
Avoid defining type alias unless truly necessary. Directly use the type. Ditto for all other type alias